src/Features/CSharp/Portable/AddParameter/CSharpAddParameterCodeFixProvider.cs

```diff
@@ -39,5 +42,40 @@ protected override ImmutableArray<string> TooManyArgumentsDiagnosticIds
 
         protected override ImmutableArray<string> CannotConvertDiagnosticIds
             => GenerateConstructorDiagnosticIds.CannotConvertDiagnosticIds;
+
+        protected override RegisterFixData<ArgumentSyntax> TryGetLanguageSpecificFixInfo(
+            SemanticModel semanticModel,
+            SyntaxNode node,
+            CancellationToken cancellationToken)
+        {
+            if (node is ConstructorInitializerSyntax constructorInitializer)
+            {
+                var constructorDeclaration = constructorInitializer.Parent;
+                if (semanticModel.GetDeclaredSymbol(constructorDeclaration, cancellationToken) is IMethodSymbol constructorSymbol)
+                {
+                    var type = constructorSymbol.ContainingType;
+                    if (constructorInitializer.IsKind(SyntaxKind.BaseConstructorInitializer))
+                    {
+                        // Search for fixable constructors in the base class.
+                        type = type?.BaseType;
+                    }
+
+                    if (type != null && type.IsFromSource())
+                    {
+                        var methodCandidates = type.InstanceConstructors;
+                        if (constructorInitializer.IsKind(SyntaxKind.ThisConstructorInitializer))
+                        {
+                            // Exclude the constructor with the diagnostic from the list of candidates
+                            methodCandidates = methodCandidates.Remove(constructorSymbol);
+                        }
```

Comment:
we could use your new helper here.  so we have a unified way of saying: do not call the containing constructor from itself.

Reply:
Moving the logic to RegisterCodeFix would break the ObjectCreation part:

class C {
 C() 
 {
    new C(1); // we want to offer the fix here.
 }
}

So we need to special case this for Me.New() and this(). For this() we have everything at hand to remove the candidate easily. For VB we need to do some more hops.

I can't really tell what is the best solution here, because all of them come with tradeoffs.

Reply:
>  // we want to offer the fix here.

Do we though?  Seems very esoteric :D

Reply:
> So we need to special case this for Me.New() and this()

Ok.  so pass in some sort of flag saying: i'm in a constructor-chaining call.  

or only do the check if we're in a constructor chaining call :)

Reply:
> Ok. so pass in some sort of flag saying: i'm in a constructor-chaining call.

That did the trick (269a68c).